posts/2022-01-14-construct-vendia-univseral-application-using-domain-driven-design.md

```diff
@@ -0,0 +1,177 @@
+</p>
+<p align="center">Figure 6 - Example Customer Uni</p>
+
+**You can use the below resources to practice and deploy pattern 2.**
+<details>
+<summary>Schema</summary>
+[schema.json](https://github.com/vendia/examples/blob/7792069dd40c6822279305b201965685aaac5a08/share/data-modeling/domain-driven-design/pattern2/schema/schema.json)
+</details>
+
+<details>
+<summary>Details and deployment instructions</summary>
+[README.md](https://github.com/vendia/examples/blob/vikrant-data-modeling-blog/share/data-modeling/domain-driven-design/pattern2/README.md)
+</details> 
+
+
+### Pattern 3: Two Different Domains, Same Bounded Context,  a Single Uni
+
+This pattern is applicable for organizations where an existing domain has its own software systems but wants to leverage the Uni owned and operated by other domains for both read and write access on a small part of the domain data model. The intent is to scale as much as on the existing Uni owned by the other domains before spinning a dedicated Uni for the domain.
+
+For Example, CRM and Marketing(Domain 2, Bounded Context 2) will need discrete write access permission boundaries between data managed by each.  You can still do it inside a single Uni. Create a node each for CRM and Marketing. You can configure write access on customer account aggregate for CRM on Node1 and configure write access on ‘CustomerProfile’ aggregate for Marketing on Node2. CRM will interact with Uni using Node 1 and Marketing will interact with Uni using Node 2. 
+
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/96793170/151029861-fb727977-4571-45f2-b719-a2b0051ad3cf.png" />
+</p>
+<p align="center">Figure 7 - Pattern 3</p>
+
+
+**You can use the below resources to practice and deploy pattern 3.**
+<details>
+<summary>Schema</summary>
+[schema.json](https://github.com/vendia/examples/blob/7792069dd40c6822279305b201965685aaac5a08/share/data-modeling/domain-driven-design/pattern3/schema/schema.json)
+</details>
+
+<details>
+<summary>Details and deployment instructions</summary>
+[README.md](https://github.com/vendia/examples/blob/vikrant-data-modeling-blog/share/data-modeling/domain-driven-design/pattern3/README.md)
+</details> 
+
+
+### Pattern 4: Multiple Domains, Multiple Bounded Contexts, and Multiple Unis
+
+This pattern is applicable to large enterprise customers with strong governance and control requirements for software owned and operated by each domain. You will also consider this pattern if the domain data model is complex, requires higher assurance and guarantee on throughput, or sensitivity of your domain justifies Uni level isolation. 
+
+ 
+
+For example, CRM and Marketing will need a dedicated Uni each because Marketing plans to bring more data from other domains that are not relevant for CRM and have different governance requirements for new data based on the sensitivity required for the newly acquired data. 
```

Comment:
```suggestion
For example, CRM and Marketing will each need a dedicated Uni because Marketing plans to bring more data from other domains that are not relevant for CRM and have different governance requirements for new data based on the sensitivity of the newly acquired data. 
```

Reply:
committed change.